class_2.tex

```diff
@@ -106,55 +119,88 @@ \subsection{Lattices and Fixpoints}
 \end{definition}
 
-\begin{example} 
-    Consider the poset $(\NN,\leq)$. Then for any $B \subseteq \NN$, if $B$ is finite, $\bigsqcup B$ is well-defined and equal to $\max B$. If $B$ is infinite, then $\bigsqcup B$ does not exist. 
+\begin{example}
+    Consider the poset $(\NN,\leq)$. Then for any $B \subseteq \NN$, if $B$ is finite, $\lub(B)$ is well-defined and equal to $\max B$. If $B$ is infinite, then $\lub(B)$ does not exist.
 \end{example}
 
-\begin{example} 
-   Consider the poset $(\NN \cup \{\infty\},\leq)$ where for all $x$ in $\NN$, it holds that $x \leq \infty$. Then for all $B \subseteq \NN$, the least upper bound $\bigsqcup B$ is well-define. 
+\begin{example}
+   Consider the poset $(\NN \cup \{\infty\},\leq)$ where for all $x$ in $\NN$, it holds that $x \leq \infty$. Then for all $B \subseteq \NN$, $\lub(B)$ is well-defined.
 \end{example}
 
-\begin{example} 
-    Let $A$ be any set and consider the poset $(\power(A),\subseteq)$. For any subset $B$ of $\power(A)$, it holds that $\bigsqcup B = \bigcup B$ and $\bigsqcap B = \bigcap B$. 
+\begin{example}
+    Let $A$ be any set and consider the poset $(\power(A),\subseteq)$. For any subset $B$ of $\power(A)$, it holds that $\lub(B) = \bigcup B$ and $\glb(B) = \bigcap B$.
 \end{example}
 
+\begin{lemma}
+  Let $(A,\sqsubseteq)$ be a poset. For all $x$, $y$ in $A$, and any subset $B$ of $A$, if $x$ and $y$ are lubs (glbs) of $B$, then $x = y$.
+\end{lemma}
+
+\begin{proof}
+    Consider arbitrary $\hat{x}, \hat{y} \in A$ and $\hat B \subseteq A$. Assume $\hat x$ is a lub of $\hat B$
+    and $\hat y$ is a lub of $\hat B$. We know:
+    \begin{enumerate}
+        \item[(i)] for all $z \in B$, $z \sqsubseteq \hat{x}$,
+        \item[(ii)] for all $z \in B$, $z \sqsubseteq \hat{y}$,
+        \item[(iii)] for all $z \in A$, if $z$ is an upper bound of $B$ then
+        $\hat x \sqsubseteq z$ and
+        \item[(iv)]for all $z \in A$, if $z$ is an upper bound of $B$ then $\hat
+        y \sqsubseteq z$.
+    \end{enumerate}
+    From (ii) we know that $\hat{y}$ is an upper bound, and therefore, by (iii), $\hat{x} \sqsubseteq \hat{y}$. Symmetrically, by (i) and (iv) we get $\hat{y} \sqsubseteq \hat{x}$. By antisymmetry, $x = y$.
+\end{proof}
+
+Given that the \emph{least upper bound} and the \emph{greatest lower bound} are unique (if they exist), for any set $A$ and $B \subseteq A$, we write $\bigsqcup B$ to denote $\lub(B)$ and $\bigsqcap B$ to denote $\glb(B)$.
+
+\begin{definition}[Lattice]
+    A poset $(A,\sqsubseteq)$ is a \emph{lattice} if for all $x$, $y$ in $A$, $\bigsqcup \{x,y\}$ ($x \sqcup y$, join) and $\bigsqcap \{x,y\}$ ($x \sqcap y$, meet) exist.
+\end{definition}
```

Comment:
maybe we can add some plain english between the comma in .`...$A$, $\bigsqcup ..`